R/update-news.R

```diff
@@ -11,21 +11,43 @@ update_news_impl <- function(messages) {
   add_to_news(news_lines)
 }
 
-collect_news <- function(messages) {
+collect_news <- function(commits) {
   if (fledge_chatty()) {
-    cli_alert("Scraping {.field {length(messages)}} commit messages.")
+    cli_alert("Digesting messages from {.field {nrow(commits)}} commits.")
   }
 
-  newsworthy_items <- messages %>%
-    gsub("\r\n", "\n", .) %>%
-    purrr::discard(~ . == "") %>%
-    purrr::map_chr(remove_housekeeping) %>%
-    purrr::map(extract_newsworthy_items) %>%
+  treat_message <- function(commit_df) {
+
+    default_newsworthy <- commit_df$message %>%
+      gsub("\r\n", "\n", .) %>%
+      purrr::discard(~ . == "") %>%
+      purrr::map_chr(remove_housekeeping) %>%
+      purrr::map(extract_newsworthy_items)
+
+    if (nrow(default_newsworthy[[1]]) > 0) {
+      return(default_newsworthy[[1]])
+    }
+
+    if (commit_df$merge) {
```

Comment:
This means always keeping the merge commit messages. At worst I suppose they're not too bad placeholders?